frappe/model/naming.py

```diff
@@ -262,7 +264,10 @@ def make_autoname(key="", doctype="", doc="", *, ignore_validate=False):
 	                DE/09/01/00001 where 09 is the year, 01 is the month and 00001 is the series
 	"""
 	if key == "hash":
-		return frappe.generate_hash(length=10)
+		# Makeshift "ULID": first 4 chars are based on timestamp, other 8 are random
+		ts = hex(struct.unpack("<Q", struct.pack("<d", time.time()))[0])
```

Comment:
This is just converting float to binary representation and picking the part that changes once _roughly_ every second.